### PR TITLE
Update Kubernetes dependencies to v1.26.0

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

- Update Kubernetes dependencies to v1.26.0
- Update controller-runtime to v0.14.1
- Update machine-controller to the latest revision (required by the controller-runtime update)
- Update the kubeone-e2e image

**Which issue(s) this PR fixes**:
xref #2561 

**What type of PR is this?**
/kind cleanup

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
NONE
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: hack/images/kubeone-e2e/Dockerfile [hunk 1]
```diff
@@ -27,12 +27,12 @@ RUN curl -fL https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terra
...
 # Source: https://github.com/vmware-tanzu/sonobuoy
-ENV SONOBUOY_VERSION "0.56.12"
+ENV SONOBUOY_VERSION "0.56.14"
 RUN curl -fL https://github.com/vmware-tanzu/sonobuoy/releases/download/v${SONOBUOY_VERSION}/sonobuoy_${SONOBUOY_VERSION}_linux_amd64.tar.gz | tar vxz
 RUN chmod +x sonobuoy
 
-# Source: https://dl.k8s.io/release/stable-1.24.txt
-ENV KUBECTL_VERSION="v1.24.8"
+# Source: https://dl.k8s.io/release/stable-1.25.txt
+ENV KUBECTL_VERSION="v1.25.5"
```
Comment (kron4eg) [inline hack/images/kubeone-e2e/Dockerfile:35]: should we use 1.26.0 here?

Comment (xmudrii) [reply hack/images/kubeone-e2e/Dockerfile:35]: Not now -- we should update this only after allowing v1.26.0 in the API validation (which will be done in a follow-up).

**Comment by kron4eg** [comment]
```
 pkg/cmd/shared.go:262:15: SA1019: sets.String is deprecated: use generic Set instead. new ways: s1 := Set[string]{} s2 := New[string]() (staticcheck)
	validSet     sets.String
	             ^ 
```

oh wow, the linter!

**Comment by xmudrii** [comment]
@kron4eg The linter error should be fixed now.

**Pull request merged**